management-node/config.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/bgzzz/go-schedule/common"
+)
+
+// Scheduling algorithms that can be used for
+// distributed scheduling
+const (
+	// SchedAlgoRR is the algorithm that equally
+	// distributes jobs/tasks among connected
+	// worker nodes
+	SchedAlgoRR = "rr"
+	// SchedAlgoRand is the algorithm that randomly
+	// selects connected worker node to execute the
+	// task/job
+	SchedAlgoRand = "rand"
+)
+
+// ServerConfig is configuration of management node
+type ServerConfig struct {
+	// BasicConfig is basic configuration part of config
+	// is is common for all parts of solution
+	BasicConfig common.Config `yaml:"basic_config"`
+
+	// Server and connection handling related configuration
+	ListeningAddress string `yaml:"listening_address"`
+	// PingTimer is period of sending ping messages to
+	// worker via stream. Management node sends ping message
+	// to worker node each PingTimer seconds
+	PingTimer int `yaml:"ping_timer"`
+	// SilenceTimeout is timeout that is used for checking
+	// silence periods during bi-directional streaming
+	// session. Server terminates stream if SilenceTimeout
+	// expired
+	SilenceTimeout int `yaml:"silence_timeout"`
+
+	// Etcd related configuration
+	// EtcdAddress is the address of db
+	EtcdAddress string `yaml:"etcd_address"`
```

Comment:
Is this a single etcd node address, or can this accept multiple nodes in some format, e.g comma-separated?

Reply:
By design of the first step, it was only one node of the etcd. I will change it in future.

Reply:
change it such that there is a reconnection timer and worker tries to reconnect if it loses the connection